src/deepsparse/transformers/pipelines_cli.py

```diff
@@ -0,0 +1,276 @@
+        f"Defaults to {default_model_path}",
+        default=default_model_path,
+        type=str,
+    )
+
+    parser.add_argument(
+        "--engine-type",
+        "--engine_type",
+        help="Inference engine name to use. Supported options are 'deepsparse'"
+        "and 'onnxruntime'",
+        type=str,
+        choices=SUPPORTED_ENGINES,
+        default=SUPPORTED_ENGINES[0],
+    )
+
+    parser.add_argument(
+        "--config",
+        help="Huggingface model config, if none provided, default will be used"
+        "which will be from the model name or sparsezoo stub if given for "
+        "model path",
+        type=str,
+        default=None,
+    )
+
+    parser.add_argument(
+        "--tokenizer",
+        help="Huggingface tokenizer, if none provided, default will be used",
+        type=str,
+        default=None,
+    )
+
+    parser.add_argument(
+        "--max-length",
+        "--max_length",
+        help="Maximum sequence length of model inputs. default is 128",
+        type=int,
+        default=128,
+    )
+
+    parser.add_argument(
+        "--num-cores",
+        "--num_cores",
+        help="Number of CPU cores to run engine with. Default is the maximum "
+        "available",
+        type=int,
+        default=None,
+    )
+    parser.add_argument(
+        "-b",
+        "--batch-size",
+        "--batch_size",
+        help="The batch size to use for pipelines",
+        type=int,
+        default=1,
+    )
+
+    scheduler_choices = ["multi", "single"]
+    parser.add_argument(
+        "--scheduler",
+        choices=scheduler_choices,
+        help="The scheduler to use for the engine. Can be None, single or multi",
+        default=scheduler_choices[0],
+    )
+
+    parser.add_argument(
+        "-o",
+        "--output-file",
+        "--output_file",
+        action="store",
+        dest="output_file",
+        type=argparse.FileType("w"),
+        help="Directs the output to a name of your choice",
+        default="-",
+    )
+
+    _args = parser.parse_args()
+    _validate_input_file(_args.data)
+    _validate_file_exists(_args.model_path)
+    return _args
+
+
+def _validate_input_file(data_file_path: str):
+    _validate_file_exists(data_file_path=data_file_path)
+    data_file = Path(data_file_path)
+    if data_file.suffix not in SUPPORTED_EXTENSIONS:
+        raise ValueError(
+            f"The input data file must be one of {SUPPORTED_EXTENSIONS}, "
+            f"found {data_file.suffix}"
+        )
+
+
+def _validate_file_exists(data_file_path: str):
+    data_file = Path(data_file_path)
+    if not data_file.exists():
+        raise ValueError(f"The file must exist, {data_file}")
+    if not data_file.is_file():
+        raise ValueError(f"Must be a file, {data_file}")
+
+
+def cli():
+    """
+    Cli entrypoint for one-shot inference using pipelines
+    """
+    _args = _parse_args()
+
+    pipe = pipeline(
```

Comment:
let's refactor this logic to an API accessible function like `pipeline_inference(pipeline, data_path, output_path)`, that way we get both the API and CLI parts of the product req